proposals/interface-types/working-notes/Instructions.md

```diff
@@ -0,0 +1,252 @@
+# Interface instructions
+
+This proposal defines new instructions that operate over interface value types. These instructions have the same form as core instructions; they push and pop values on an execution stack and support single pass validation.
+
+The new interface instructions are summarized below.
+
+```
+#interface-instr ::=
+  | bool.lift_iMM
+  | bool.lower_iMM
+
+  | sNN.lift_iMM
+  | sNN.lower_iMM
+
+  | uNN.lift_iMM
+  | uNN.lower_iMM
+
+  | string.lift_memory ..
+  | string.lower_memory ..
+
+  | record.lift ..
+  | record.lower ..
+
+  | array.lift_memory ..
+  | array.lower_memory ..
+
+  | invoke ..
+
+where:
+  NN is one of [8, 16, 32, 64]
+  MM is one of [32, 64]
+```
+
+In addition the following instructions are included from the core instructions:
+
+```
+#interface-instr ::=
+  | ..
+  | <numeric instructions>
+  | <memory instructions>
+  | <table instructions>
+  | drop
+  | unreachable
+  | nop
```

Comment:
I believe we will need essentially all of core wasm instructions.
Certainly we need all the non-control flow instructions

Reply:
it is a bit soon IMO to be definitive about which instructions from core wasm we will need.
so far, I have found that every scenario I look at has the effect of bringing in more.
For example, to handle exceptions we will likely need to bring all of that in -- even if IT itself does not have exceptions.